### PR TITLE
Dmitry/init

### PR DESCRIPTION
**Pull request by dima-engineer**
Init PR

### CODE REVIEW

File: snippets/example.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+from dotenv import load_dotenv
+from pydantic import BaseModel, Field
+from pydantic_settings import BaseSettings, PydanticBaseSettingsSource
+
+from pydantic_secrets import SecretManagerClientABC, SecretManagerSource
+
+
+class SecretManagerExample(SecretManagerClientABC):
```
Comment (shakun-roman) [inline snippets/example.py:8]: From my point of view realization of this manager  should be inside of library.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+line-length = 120
+src = ["./"]
+extend-select = [
+    "I",
+    "F",
+    "N",
+    "UP",
+]
+
+#[tool.pyright]
```
Comment (shakun-roman) [inline pyproject.toml:33]: remove or uncomment

**Review by shakun-roman — CHANGES_REQUESTED** [review]

File: pydantic_secrets/main.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        self.__secret_manager__ = secret_manager_client
+        self.__default_secret_version__ = default_secret_version
+
+    def get_field_value(
+        self, field: FieldInfo, field_name: str
+    ) -> tuple[Any, str, bool]:
+        return field_name, field.default, self.field_is_complex(field)
+
+    def __call__(self) -> dict[str, Any]:
+        return self._process_fields(self.settings_cls.model_fields)
```
[the remark this answers is not in the document]
Comment (dima-engineer) [reply pydantic_secrets/main.py:37]: To be able to run it recursively

File: pydantic_secrets/main.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            )
+            if secret_name and secret_version:
+                try:
+                    d[field_name] = self.__secret_manager__.get_secret(
+                        secret_name=secret_name,
+                        secret_version=secret_version,
+                    )
+                except Exception:
+                    warnings.warn(field_name, category=SecretReceivingWarning)
+                    pass
```
[the remark this answers is not in the document]
Comment (dima-engineer) [reply pydantic_secrets/main.py:76]: It's done for cases when you want the secret manager source to be not the only source. So, it's still possible to load variable from other sources.

File: lets.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+env:
+  GREEN: '\033[0;32m'
+  CYAN: '\033[0;36m'
+  YELLOW: '\033[0;33m'
+  CLEAR: '\033[0m'
+  SRC_PATH: './'
+
+commands:
+  format:
+    description: Run code auto formatting tools tools
```
[the remark this answers is not in the document]
Comment (dima-engineer) [reply lets.yaml:12]: ruff covers it

File: snippets/example.py (hunk 1 above)
Comment (dima-engineer) [reply snippets/example.py:8]: This is just a code snippet - example of usage

Comment (dima-engineer) [reply snippets/example.py:8]: I took [this](https://github.com/googleapis/google-api-python-client) as example

File: pydantic_secrets/main.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    def __call__(self) -> dict[str, Any]:
+        return self._process_fields(self.settings_cls.model_fields)
+
+    def _get_secret_info_from_extra(
+        self,
+        extra: dict[str, t.Any],
+    ) -> tuple[str | None, str | None]:
+        secret_name = extra.get(EXTRA_SECRET_NAME_KEY)
+        secret_name_env = extra.get(ENV_SECRET_NAME_KEY)
+        if secret_name_env:
```
[the remark this answers is not in the document]
Comment (dima-engineer) [reply pydantic_secrets/main.py:45]: Let's say that I have a secret with the same name "my-secret" in stage/prod but in dev it's called "my-secret-bla-bla-bla". So, I can use default `secret_name="my-secret"` for stage and prod (I don't define env variable in these envs). But in dev I can define env variable to rewrite the secret name to `my-secret-bla-bla-bla`

File: pydantic_secrets/main.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+EXTRA_SECRET_VERSION_KEY = "secret_version"
+ENV_SECRET_VERSION_KEY = "secret_version_env"
+
+
+class SecretManagerSource(PydanticBaseSettingsSource):
+    def __init__(
+        self,
+        settings_cls: type[BaseSettings],
+        secret_manager_client: SecretManagerClientProtocol,
+        default_secret_version: str | None = None,
```
[the remark this answers is not in the document]
Comment (dima-engineer) [reply pydantic_secrets/main.py:25]: I think that the current approach makes the library easier to use. If I'm gonna specify secret_version for each secret I will define, I don't need default_secert_version. And currently, the library doesn't enforce you to specify this parameter if you don't need it

File: pydantic_secrets/main.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (shakun-roman) [reply pydantic_secrets/main.py:70]: i mean remove `pass`, we don't need it

File: pydantic_secrets/main.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (shakun-roman) [reply pydantic_secrets/main.py:41]: i mean that lib should mimic pydantic behaviour. In pydantic an explicitly specified value has a priority over env vars

[the remark this answers is not in the document]
Comment (dima-engineer) [reply pydantic_secrets/main.py:41]: But in pydantic-settings it's reversed. If you specify default value for a field, but there is a env variable specified - ENV value will be over the default value

File: snippets/example.py (hunk 1 above)
Comment (dima-engineer) [reply snippets/example.py:8]: Will be implemented in another PR

File: pydantic_secrets/main.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (shakun-roman) [reply pydantic_secrets/main.py:41]: got it

**Review by shakun-roman — APPROVED** [review]

**Pull request merged**